File: Golang/longestContinuousSubArray.go
// Solution: keep track of the previous element, if the current > previous increment the counter
// Runtime: O(n)

func findLengthOfLCIS(nums []int) int {

    if nums == nil || len(nums) == 0 {
        return 0
    }

    prev := nums[0]
    max := 1
    counter := 1

    for i := 1; i < len(nums); i++ {
        if nums[i] > prev {
            counter++
            prev = nums[i]
        } else {
            counter = 1
            prev = nums[i]
        }

        if counter > max {
            max = counter
        }
    }
    return max
}

File: Easy/Java/longestContinuousSubArray.java
// Solution: keep track of the previous element, if the current > previous increment the counter
// Runtime: O(n)

class Solution {
    public int findLengthOfLCIS(int[] nums) {

        if (nums == null || nums.length == 0) {
            return 0;
        }

        int counter = 1;
        int max = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i-1]) {
                counter++;
            }
            else {
                counter = 1;
            }

            if (counter > max) {
                max = counter;
            }
        }
        return max;
    }
}
